### PR TITLE
PTEUDO-1086 provision alloydb instance

### PR DESCRIPTION
**Pull request by EvertonCalgarotto**

### CODE REVIEW

File: test/e2e/dbc_end2end_test.go [hunk 1]
```diff
@@ -49,12 +52,12 @@ var (
 	db2                    string
 	db3                    string
 	ctx                    = context.Background()
+	cloud                  string
 )
 
-var _ = Describe("AWS", Ordered, func() {
+var _ = Describe("AWS/GCP", Ordered, func() {
```
Comment (drewwells) [inline test/e2e/dbc_end2end_test.go:58]: I like the optimism that the same code will work in both clouds :D

Comment (EvertonCalgarotto) [reply test/e2e/dbc_end2end_test.go:58]: huahauuah, trust the process!!

File: helm/db-controller/templates/cluster-role.yaml [hunk 2]
```diff
@@ -63,6 +63,108 @@ rules:
...
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
+  - apiGroups:
+    - persistance.infoblox.com
+    resources:
+    - xnetworkrecords
```
Comment (drewwells) [inline helm/db-controller/templates/cluster-role.yaml:95]: Hopefully, these non-existent CRD references will not result in errors on clusters.

File: helm/db-controller/values.yaml [hunk 3]
```diff
@@ -132,8 +132,8 @@ controllerConfig:
   athena-shared:
     masterUsername: root
   authSource: secret
-  # if aws authorization is used iam role must be provided
-  #iamRole: rds-role
+  #aws or gcp
+  cloud: aws
```
Comment (drewwells) [inline helm/db-controller/values.yaml:136]: This isn't configured in DC. TODO to fix this in a later PR
```suggestion
  # FIXME: change me to `cloud_provider` and update DC kubevela workflow to populate the field
  # https://github.com/Infoblox-CTO/deployment-configurations/pull/61446/files#diff-796b260c9364c2558037d151b9450f8b72904b41ccb90b2f1da7aff36e97e141R9-R24
  cloud: aws
```

File: cmd/config/config.yaml [hunk 4]
```diff
@@ -1,13 +1,10 @@
-# master credentials source can be 'aws' or 'secret'
-#authSource: aws
+athena-shared:
+  masterUsername: root
 authSource: secret
+#aws or gcp
+cloud: aws
```
Comment (drewwells) [inline cmd/config/config.yaml:5]: Do I have to edit this to say `gcp` to run integration tests?

File: config/crd/bases/persistance.infoblox.com_xnetworkrecords.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
+apiVersion: apiextensions.crossplane.io/v1
```
Comment (drewwells) [inline config/crd/bases/persistance.infoblox.com_xnetworkrecords.yaml:1]: Really wish kubebuilder would let us build rbac without generating CRD. This is fine here, but somebody people are going to break their cluster installing this CRD.

File: pkg/basefunctions/basefunctions.go [hunk 6]
```diff
@@ -190,10 +191,34 @@ func GetIsPasswordComplexity(viperConfig *viper.Viper) bool {
...
+	return viperConfig.GetString("cloud")
+}
+
 func GetRegion(viperConfig *viper.Viper) string {
 	return viperConfig.GetString("region")
 }
 
+// GCP only
+func GetNetwork(viperConfig *viper.Viper) string {
+	return fmt.Sprintf("projects/%s/global/networks/%s", GetProject(viperConfig), viperConfig.GetString("network"))
```
Comment (drewwells) [inline pkg/basefunctions/basefunctions.go:204]: We probably should not try to rebuild fully qualified network and subnet names in the config loader function. Let's just require that db-controller is configured with the fully qualified name in the terraform ticket ie.
`/projects/gcp-eng-ddiaas-dev/global/networks/ddiaas-dev-use1-vpc`

File: scripts/psql-open-owners.sh [hunk 7]
```diff
@@ -22,8 +22,9 @@ get_table_owners() {
     fi
     printf "\nNamespace: $namespace\nDatabaseClaim: $claim_name\n"
     printf "Created: $(kubectl get databaseclaim "$claim_name" -n "$namespace" -o jsonpath='{.metadata.creationTimestamp}')\n"
+    printf "Secret: $(kubectl get databaseclaim "$claim_name" -n "$namespace" -o jsonpath='{.spec.secretName}')\n"
     # Run psql command to find the owner of tables
-    kubectl exec deploy/db-controller -n db-controller -c manager -- psql "$dsn" -c '\dt' \
+    kubectl exec deploy/db-controller -n db-controller -c postgres -- psql "$dsn" -c '\dt' \
```
Comment (drewwells) [inline scripts/psql-open-owners.sh:27]: looks like manager to me https://github.com/infobloxopen/db-controller/blob/699abc587578222c20e3efb77c490ecdbc34300c/helm/db-controller/templates/deployment.yaml#L32

**Review by drewwells — APPROVED** [review]

File: cmd/config/config.yaml (hunk 4 above)
Comment (EvertonCalgarotto) [reply cmd/config/config.yaml:5]: yes, this key and the region below, cause GCP uses us-east1, without the dash before the number 1

File: pkg/basefunctions/basefunctions.go (hunk 6 above)
Comment (EvertonCalgarotto) [reply pkg/basefunctions/basefunctions.go:204]: you mean to use the full name in a single key?

**Pull request merged**